### DISCUSSION TITLE
Best practice for number of srcset items?

### DISCUSSION THREAD

**Discussion by nathanchase** [nuxt/image#728] [Q&A]
This is more of an optimization question, but how many `scrset` items is too many? 

![image](https://user-images.githubusercontent.com/5963574/215290904-7b7b4f03-f8f4-4cb0-8028-6640dea2767c.png)

With @nuxt/image we can have infinite sizes since IPX can resize on demand... but when does it become too over-optimized?

Is it too taxing/overburdening to ask IPX to make so many specific sizes for an image? I can't find much online in the direction of ideal scenarios for what to put in a `srcset`.

**Comment by danielpetrica** [comment]
It's jut a personal thing but I think images under 100w could use just the 100w. 
For the other sizes maybe using just the standard image sizes could be a fast solution? 
To be clear I'm thinking about the named youtube resolutions like 240, 360, 480, 720, 1080 and so on

**Comment by DavidDeSloovere** [comment]
As I see this, I wonder how dangerous it is to allow dynamic resizing/generation of every width x heigth combination. This could generate more images than you hosting (bill) could handle.

**Reply by nathanchase** [reply]
Well, these are dynamically generated, they're not taking up storage per se. The goal is to send down the wire the smallest possible image for any given circumstance... the question is more whether it computationally complicates the DOM in a detrimental way with so many possible permutations.

**Reply by DavidDeSloovere** [reply]
If they are not taking up space, then it will take up memory and CPU. 
Enumeration all urls as part of an attack seems likely to me. Without safeguards on allowed variations (ie. dimensions need to be mutiples of 10) an image of 1000x1000 pixels has almost 1.000.000 combinations.

**Reply by DavidDeSloovere** [reply]
Also, I'm not talking about use devs referencing too many images in our code.
But attackers just calling the IPX endpoint directly.

**Reply by nathanchase** [reply]
Sure, I suppose someone could try to DDOS a site running sharp or ipx and hit randomized size images programmatically and hammer the server as a result.
